File: PairingCreater/src/main/java/org/dongguk/crewpairing/domain/Flight.java
package org.dongguk.crewpairing.domain;

import lombok.*;
import org.dongguk.common.domain.AbstractPersistable;

import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;

@Getter
@Setter
@AllArgsConstructor
public class Flight extends AbstractPersistable {
    private String serialNumber;
    //비행편 T/N
    private String tailNumber;
    //출발 공항
    private Airport originAirport;
    //출발 시간
    private LocalDateTime originTime;
    //도착 공항
    private Airport destAirport;
    //도착 시간
    private LocalDateTime destTime;
    //기종
    private Aircraft aircraft;
    //운항시간
    private int flightTime;

    @Override
    public String toString() {
        return "F"+ id;
    }

    @Builder
    public Flight(long id, String serialNumber, String tailNumber,
                  Airport originAirport, LocalDateTime originTime,
                  Airport destAirport, LocalDateTime destTime, Aircraft aircraft) {
        super(id);
        this.serialNumber = serialNumber;
        this.tailNumber = tailNumber;
        this.originAirport = originAirport;
        this.originTime = originTime;
        this.destAirport = destAirport;
        this.destTime = destTime;
        this.aircraft = aircraft;
        this.flightTime = (int) ChronoUnit.MINUTES.between(originTime, destTime);
    }
}

File: ReinforcementLearning/Flight.py
from datetime import datetime
from datetime import timedelta

class Flight:
    def __init__(self, id, TailNumber, originAirport, originTime, destAirport, destTime, aircraft):
        self.id = id
        self._TailNumber = TailNumber
        self._originAirport = originAirport
        self._originTime = originTime
        self._destAirport = destAirport
        self._destTime = destTime
        self._aircraft = aircraft
        self._flightTime = int((self._destTime - self._originTime).total_seconds() // 60)
    
    @property
    def TailNumber(self):
        return self._TailNumber
    
    @TailNumber.setter
    def TailNumber(self, value):
        self._TailNumber = value

    @property
    def originAirport(self):
        return self._originAirport
    
    @originAirport.setter
    def originAirport(self, value):
        self._originAirport = value

    @property
    def originTime(self):
        return self._originTime

    @originTime.setter
    def originTime(self, value):
        self._originTime = value

    @property
    def destAirport(self):
        return self._destAirport
    
    @destAirport.setter
    def destAirport(self, value):
        self._destAirport = value

    @property
    def destTime(self):
        return self._destTime
    
    @destTime.setter
    def destTime(self, value):
        self._destTime = value

    @property
    def aircraft(self):
        return self._aircraft
    
    @aircraft.setter
    def aircraft(self, value):
        self._aircraft = value

    @property
    def flightTime(self):
        return self._flightTime

    def getIndex(self):
        return "F" + str(self.id)

    def __str__(self):
        return "F" + str(self.id)
